File: ChefEnCasa/soap-net/App_Code/Models/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

public class Message
{
        public int IdMessage { get; set; }

        public int SenderId { get; set; }

        public User Sender { get; set; }

        public int ReceiverId { get; set; }

        public User Receiver { get; set; }

        public string Subject { get; set; }

        public string Value { get; set; }

        public string Response { get; set; }

        public bool Answered { get; set; }
}

File: ChefEnCasa/grpc-java-server/src/main/java/com/chefencasa/Model/Message.java
package com.chefencasa.Model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@Table(name="messages")
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idMessage")
    private int idMessage;

    @ManyToOne (cascade={CascadeType.ALL, CascadeType.REFRESH})
    @JoinColumn (name = "idSender")
    private User sender;

    @ManyToOne (cascade={CascadeType.ALL, CascadeType.REFRESH})
    @JoinColumn (name = "idReceiver")
    private User receiver;

    @Column(name="subject", nullable = false)
    private String subject;

    @Column(name="value", nullable = false)
    private String value;

    @Column(name="response")
    private String response;

    @Column(name="answered")
    private boolean Answered;
}
